x/distribution/keeper/allocation_test.go

```diff
@@ -106,3 +106,74 @@ func TestAllocateTokensToManyValidators(t *testing.T) {
 	// proposer reward + staking.proportional for second proposer = (5 % + 0.5 * (93%)) * 100 = 51.5
 	require.Equal(t, sdk.DecCoins{{sdk.DefaultBondDenom, sdk.NewDecWithPrec(515, 1)}}, k.GetValidatorCurrentRewards(ctx, valOpAddr2).Rewards)
 }
+
+func TestAllocateTokensTruncation(t *testing.T) {
+	communityTax := sdk.NewDec(0)
+	ctx, _, k, sk, fck := CreateTestInputAdvanced(t, false, 1000000, communityTax)
+	sh := staking.NewHandler(sk)
+
+	// initialize state
+	k.SetOutstandingRewards(ctx, sdk.DecCoins{})
+
+	// create validator with 50% commission
+	commission := staking.NewCommissionMsg(sdk.NewDecWithPrec(1, 1), sdk.NewDecWithPrec(1, 1), sdk.NewDec(0))
+	msg := staking.NewMsgCreateValidator(valOpAddr1, valConsPk1,
+		sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(110)), staking.Description{}, commission, sdk.OneInt())
+	require.True(t, sh(ctx, msg).IsOK())
+
+	// create second validator with 0% commission
+	commission = staking.NewCommissionMsg(sdk.NewDecWithPrec(1, 1), sdk.NewDecWithPrec(1, 1), sdk.NewDec(0))
+	msg = staking.NewMsgCreateValidator(valOpAddr2, valConsPk2,
+		sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100)), staking.Description{}, commission, sdk.OneInt())
+	require.True(t, sh(ctx, msg).IsOK())
+
+	// create second validator with 0% commission
+	commission = staking.NewCommissionMsg(sdk.NewDecWithPrec(1, 1), sdk.NewDecWithPrec(1, 1), sdk.NewDec(0))
+	msg = staking.NewMsgCreateValidator(valOpAddr3, valConsPk3,
+		sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100)), staking.Description{}, commission, sdk.OneInt())
+	require.True(t, sh(ctx, msg).IsOK())
+
+	abciValA := abci.Validator{
+		Address: valConsPk1.Address(),
+		Power:   11,
+	}
+	abciValB := abci.Validator{
+		Address: valConsPk2.Address(),
+		Power:   10,
+	}
+	abciValС := abci.Validator{
+		Address: valConsPk3.Address(),
+		Power:   10,
+	}
+
+	// assert initial state: zero outstanding rewards, zero community pool, zero commission, zero current rewards
+	require.True(t, k.GetOutstandingRewards(ctx).IsZero())
+	require.True(t, k.GetFeePool(ctx).CommunityPool.IsZero())
+	require.True(t, k.GetValidatorAccumulatedCommission(ctx, valOpAddr1).IsZero())
+	require.True(t, k.GetValidatorAccumulatedCommission(ctx, valOpAddr2).IsZero())
+	require.True(t, k.GetValidatorCurrentRewards(ctx, valOpAddr1).Rewards.IsZero())
+	require.True(t, k.GetValidatorCurrentRewards(ctx, valOpAddr2).Rewards.IsZero())
+
+	// allocate tokens as if both had voted and second was proposer
+	fees := sdk.Coins{
+		{sdk.DefaultBondDenom, sdk.NewInt(634195840)},
```

Comment:
Would prefer to use the `NewCoin` constructor here (even thought he input is valid).

Reply:
Updated